app/bundles/LeadBundle/Model/LeadModel.php

```diff
@@ -1184,6 +1175,27 @@ public function import($fields, $data, $owner = null, $list = null, $tags = null
         $lead ??= $this->checkForDuplicateContact($fieldData);
         $merged = (bool) $lead->getId();
 
+        // Extract company data and import separately
+        // Modifies the data array
+        $company                       = null;
+        [$companyFields, $companyData] = $this->companyModel->extractCompanyDataFromImport($fields, $data);
+
+        if ((true === $skipIfExists && empty($lead->getCompany())) || false === (bool) $skipIfExists) {
```

Comment:
Is it intentional that when checking for `false` then the variable is retyped (`(bool) $skipIfExists`) but when checking for `true` it isn't? It doesn't seem right. Either both need it or neither, right?